### ISSUE TITLE
Fusion: Define endpoint in an interface

### ISSUE THREAD

**Issue by fluorumlabs** [vaadin/flow#9504]
As a developer I want to define Fusion endpoint in an interface so that I can supply different implementations using Spring facilities (for example, use dummy data provider in development mode)

**Comment by Legioth** [comment]
As a workaround, you can use the same endpoint implementation and inject different implementations of the data provider into it.

With some additional boilerplate, you can even let the endpoint implementation be an empty shell that just delegates everything directly to an injected implementation.

**Comment by fluorumlabs** [comment]
Of course. But Fusion is all about reducing amount of boilerplate, so...
